components/search/backend/src/main/kotlin/SearchService.kt

```diff
@@ -30,90 +32,124 @@ import org.eclipse.apoapsis.ortserver.dao.repositories.ortrun.OrtRunsTable
-
-        // Build base query
-        var query = OrtRunsTable
-            .innerJoin(AnalyzerJobsTable, { OrtRunsTable.id }, { ortRunId })
-            .innerJoin(AnalyzerRunsTable, { AnalyzerJobsTable.id }, { analyzerJobId })
-            .innerJoin(PackagesAnalyzerRunsTable, { AnalyzerRunsTable.id }, { PackagesAnalyzerRunsTable.analyzerRunId })
-            .innerJoin(PackagesTable, { PackagesAnalyzerRunsTable.packageId }, { PackagesTable.id })
-            .innerJoin(IdentifiersTable, { PackagesTable.identifierId }, { IdentifiersTable.id })
-
-        // Convert Identifier to a concatenated string format for ILike comparison
-        val concatenatedIdentifier = concat(
-            IdentifiersTable.type,
-            stringLiteral(":"),
-            IdentifiersTable.namespace,
-            stringLiteral(":"),
-            IdentifiersTable.name,
-            stringLiteral(":"),
-            IdentifiersTable.version
+        userId: String,
+        scope: HierarchyId? = null
+    ): List<RunWithPackage> {
+        val hierarchyFilter = authorizationService.filterHierarchyIds(
+            userId = userId,
+            repositoryPermissions = setOf(RepositoryPermission.READ),
+            containedIn = scope
         )
 
-        val conditions = mutableListOf(concatenatedIdentifier.applyRegex(identifier))
-
-        val scopeRequested = organizationId != null || productId != null || repositoryId != null
-        if (scopeRequested) {
-            query = query
+        return db.blockingQuery {
+            val query = OrtRunsTable
+                .innerJoin(AnalyzerJobsTable, { OrtRunsTable.id }, { ortRunId })
+                .innerJoin(AnalyzerRunsTable, { AnalyzerJobsTable.id }, { analyzerJobId })
+                .innerJoin(PackagesAnalyzerRunsTable, { AnalyzerRunsTable.id }, { analyzerRunId })
+                .innerJoin(PackagesTable, { PackagesAnalyzerRunsTable.packageId }, { PackagesTable.id })
+                .innerJoin(IdentifiersTable, { PackagesTable.identifierId }, { IdentifiersTable.id })
                 .innerJoin(RepositoriesTable, { OrtRunsTable.repositoryId }, { RepositoriesTable.id })
                 .innerJoin(ProductsTable, { RepositoriesTable.productId }, { ProductsTable.id })
-        }
 
-        organizationId?.let { conditions += ProductsTable.organizationId eq it }
-        productId?.let { conditions += RepositoriesTable.productId eq it }
-        repositoryId?.let { conditions += OrtRunsTable.repositoryId eq it }
+            val concatenatedIdentifier = concat(
+                IdentifiersTable.type,
+                stringLiteral(":"),
+                IdentifiersTable.namespace,
+                stringLiteral(":"),
+                IdentifiersTable.name,
+                stringLiteral(":"),
+                IdentifiersTable.version
+            )
 
-        val whereClause = conditions.reduce { acc, expression -> acc and expression }
+            val scopeCondition = when (scope) {
```

Comment:
I think, this condition should not be required. The condition generated by the `HierarchyFilter` should already include exactly the hierarchy elements that are visible for the user.